test/serializer/basic/CapturedScope.js

```diff
@@ -1,4 +1,5 @@
 // serialized function clone count: 0
+// does not contain:__scope_2_unique27277, __scope_2_unique27277
```

Comment:
This is not ideal to test the change: it relies on the test runner to use "unique27277" as suffix which may be changed later.
You may want to use "// Copies of" or find another stable pattern as test invariant.

Reply:
Mhh, but we need to know the name of the scope somehow - regexes are not supported right now I think.
The 27277 seems to hard coded in the tester script anyway:
![image](https://user-images.githubusercontent.com/6901762/33681331-063025e4-da79-11e7-96e1-6facb85c8807.png)

Reply:
The second iteration of the test will use a different suffix and the "does not contain" directive will be inoperative. The copies of alternative invokes a path in the runner that does not add the suffix and does not iterate to a fixed point.

Reply:
For `copies of` I need at least one match though, so `copies of __scope_2, __scope_2:0` doesn't work and `copies of __scope_2:17` is also kinda confusing.

Reply:
I suggest you create a new testcase using the simplified testcase from @simonhj. 
Then you can check // Copies of __scope_1: 1. 
(Actually "__scope_1" function should be called "__get_scope_bindings", but there is another bug that incorrect named it).

Reply:
We would need `// Copies of __scope_2:6` but then this test could easily break when someone changes the code generation.

```javascript
function f (x) {
  var valueA = 0;
  function c() {
    x ? valueA++ : valueA += 2;
    return valueA;
  }

  return c;
}

var s = f(true);
var r = f(false);
```

Reply:
Well then, I am persuaded that your test is fine as it is now. :-)